src/firefly/java/edu/caltech/ipac/table/io/VoTableWriter.java

```diff
@@ -114,6 +114,29 @@ private String xmlDESCRIPTION() {
             return isEmpty(descVal) ? "" : "<"+tagDesc+">" + descVal + "</"+tagDesc+">";
         }
 
+        private String xmlTABLE() {
+            long nrow = dataGroup.size();
+            String tname = dataGroup.getTableMeta().getAttribute(TableMeta.NAME);
+            String res = "<TABLE";
```

Comment:
I noticed you are not using `StringBuilder` to do that concatenations. I also noticed that my IntelliJ was not complaining either. This got me wondering if it is still necessary in java 18. It seems that it is not required in cases like this.

I found this article: https://dzone.com/articles/string-concatenation-performacne-improvement-in-ja
https://dzone.com/articles/string-concatenation-performacne-improvement-in-ja

So the `+=` is fine.

Reply:
this is good to know, thank you!